TombEngine/Game/effects/effects.cpp

```diff
@@ -874,15 +935,61 @@ void TriggerExplosionBubbles(int x, int y, int z, short roomNumber)
 	{
 		auto* spark = GetFreeParticle();
 
-		spark->sR = 128;
-		spark->dR = 128;
-		spark->dG = 128;
-		spark->dB = 128;
+		if (mainColor == Vector3::Zero)
+		{
+			spark->sR = 128;
+			spark->sG = 64;
+			spark->sB = 0;
+			spark->dR = 128;
+			spark->dG = 128;
+			spark->dB = 128;
+		}
+		else
+		{
+			// New colored flame processing.
```

Comment:
Why this comment is saying something about "colored flame"? It indicates unchecked copypaste.